modules/json/src/lib/parse-json-in-batches.js

```diff
@@ -32,8 +33,8 @@ export default async function* parseJSONInBatches(asyncIterator, options) {
     }
 
     // Add the row
-    for (const row of rows) {
-      tableBatchBuilder.addRow(row);
+    for (let i = 0; i < rows.length; i++) {
+      tableBatchBuilder.addRow(rows[i], cursors[i]);
```

Comment:
Do we need a cursor for every row? Isn't it enough per batch? Does this enable some significantly better UX?

If we add such a feature, I would prefer that it worked for all batched loaders so I don't want to make it too complicated.

Reply:
batches are convert to rows and sent to `RowTableBatch`, rows are send back to app only when `tableBatchBuilder.isFull`. 1 batch can be broken down into multiple rows batches, or 1 row batches can consist of multiple batches depends on the `batchSize` prop of `RowTableBatch`, which is NOT the btye size, it is the number of rows. There is no way to convert the btye size of the batch to the actual btye size of the rows received by the application.

Unless you calculate avg byte size per row using `avgByteRow = chunk.length / rows.length` yet you still need a way to pass that size per row to `tableBatchBuilder.addRow` or `tableBatchBuilder.getNormalizedBatch` so that when `TableBatchBuilder` send the rows back, it calculated size using `row.length * avgByteRow`

Reply:
OK I understand now. 

Let me ask jow precise does it need to be though. could it just be an approximation, enough to reasonably feed a progress bar?

One reason I am "pushing back" is that we have multiple batch converters, including `ColumnarBatch` and `ArrowBatch` - this will spread quickly if it needs to be supported

Reply:
What about returning the bytes per record or per batch from the streaming parser, instead of the number of rows? I'm guessing when a new file is added you can get the total number of bytes. If each batch were able to contain some sort of metadata referring to its byte length, you could use that to infer the percentage of bytes that have been parsed.